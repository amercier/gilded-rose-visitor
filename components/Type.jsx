import React from 'react';
import { string } from 'prop-types';
import Chip from '@material-ui/core/Chip';

const TYPE_COLORS = {
  STANDARD: 'default',
  BACKSTAGE_PASS: 'primary',
  CONJURED: 'secondary',
};

const TYPE_LABELS = {
  STANDARD: 'Standard',
  BACKSTAGE_PASS: 'Backstage pass',
  CONJURED: 'Conjured',
};

/**
 * A simple chip that displays.
 *
 * @param {string} type - The type to display.
 * @returns {React.Element} The rendered element.
 */
const Type = ({ type }) => (
  <Chip label={TYPE_LABELS[type]} color={TYPE_COLORS[type] || 'default'} />
);

Type.propTypes = {
  type: string.isRequired,
};

export default Type;
