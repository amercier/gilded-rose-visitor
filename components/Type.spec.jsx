import React from 'react';
import { shallow } from 'enzyme';
import Type from './Type';

describe('Layout', () => {
  it('renders a STANDARD type', () => {
    expect(shallow(<Type type="STANDARD" />)).toMatchSnapshot();
  });

  it('renders a BACKSTAGE_PASS type', () => {
    expect(shallow(<Type type="BACKSTAGE_PASS" />)).toMatchSnapshot();
  });

  it('renders a CONJURED type', () => {
    expect(shallow(<Type type="CONJURED" />)).toMatchSnapshot();
  });
});
