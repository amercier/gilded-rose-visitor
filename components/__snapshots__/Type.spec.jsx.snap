// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders a BACKSTAGE_PASS type 1`] = `
<WithStyles(Chip)
  color="primary"
  label="Backstage pass"
/>
`;

exports[`Layout renders a CONJURED type 1`] = `
<WithStyles(Chip)
  color="secondary"
  label="Conjured"
/>
`;

exports[`Layout renders a STANDARD type 1`] = `
<WithStyles(Chip)
  color="default"
  label="Standard"
/>
`;
